packages/headless/src/modules/order/services/customer.service.ts

```diff
@@ -32,6 +37,40 @@ export class OrderCustomerService {
     );
   }
 
+  async reOrder(userId: string, body: ReOrderQuery): Promise<any>{
+    let { ignoreInvalidItems, overWriteCart, orderId } = body; 
+
+    const order = await this.orderRepository.getOrderById(orderId);
+    const products = order.products;
+    
+    const unavailableProducts = await this.orderRepository.getAvailableProducts(products);
+    if(unavailableProducts.length === products.length)return errorResponse( 'Products are not available right now',null,HttpStatus.BAD_REQUEST );
+    else {
+        let availableAddToCart;
+        const allAddToCart = products.map(product => { return {productId: product.productId, quantity: product.quantity}});
+
+        if(unavailableProducts.length > 0){
+            if(ignoreInvalidItems === false) return errorResponse( 'All of the products are not available. Do you wish to continue?', null, HttpStatus.BAD_REQUEST );
```

Comment:
keep meaningful enum in error response so that front-end can check that enum to precisely know what error has been occurred.